runtime/runtime/src/lib.rs

```diff
@@ -1713,64 +1789,92 @@ impl Runtime {
         let apply_state = &mut processing_state.apply_state;
         let state_update = &mut processing_state.state_update;
 
-        let signed_txs =
-            processing_state.transactions.transactions.into_iter().cloned().collect::<Vec<_>>();
-        for (tx_hash, result) in Self::parallel_validate_transactions(
-            &apply_state.config,
-            apply_state.gas_price,
-            signed_txs,
-            apply_state.current_protocol_version,
-        ) {
-            match result {
-                Ok((validated_tx, cost)) => {
-                    let (receipt, outcome_with_id) = match self.process_transaction(
-                        state_update,
-                        apply_state,
-                        &validated_tx,
-                        &cost,
-                        &mut processing_state.stats,
-                    ) {
-                        Ok(outcome) => outcome,
-                        Err(err) => {
-                            Self::handle_invalid_transaction(
-                                err,
-                                &tx_hash,
-                                processing_state.protocol_version,
-                                "process_transaction error",
-                            )?;
-                            continue;
-                        }
-                    };
-                    if receipt.receiver_id() == validated_tx.signer_id() {
-                        processing_state.local_receipts.push_back(receipt);
-                    } else {
-                        receipt_sink.forward_or_buffer_receipt(
-                            receipt,
-                            apply_state,
-                            state_update,
-                            processing_state.epoch_info_provider,
-                        )?;
+        let tx_groups = TransactionGroups::new(processing_state.transactions.transactions);
+
+        let group_outcomes: Vec<Result<GroupResult, RuntimeError>> = tx_groups
+            .par_grouped()
+            .map(|group| {
+                self.verify_grouped_transactions(
+                    group,
+                    &apply_state,
+                    state_update,
+                    apply_state.gas_price,
+                    apply_state.block_height,
+                    apply_state.current_protocol_version,
+                    None,
+                )
+            })
+            .collect();
```

Comment:
If you do the following, you can remove the type specification above.

```suggestion
            .collect::<Vec<_>>();
```

Reply:
Even better, use [`collect_vec_list`](https://docs.rs/rayon/latest/rayon/iter/trait.ParallelIterator.html#method.collect_vec_list) and flatten results during iteration.

Reply:
`IndexedParallelIterator` is not available for `ChunkBy`, and we still need to sort them batch results by original chunk order so it's better to stick with `collect::<Vec>`